packages/sitecore-jss-dev-tools/src/package-deploy.ts

```diff
@@ -305,39 +318,34 @@ export async function packageDeploy(options: PackageDeployOptions) {
   formData.append('path', fs.createReadStream(packageFile));
   formData.append('appName', options.appName);
 
-  const isHttps = options.importServiceUrl.startsWith('https');
-
   const requestBaseOptions = {
-    transport: isHttps ? getHttpsTransport(options) : undefined,
     headers: {
       'User-Agent': 'Sitecore/JSS-Import',
       'Cache-Control': 'no-cache',
       'X-JSS-Auth': hmac(factors, options.secret),
       ...formData.getHeaders(),
     },
-    proxy: extractProxy(options.proxy),
-    httpsAgent: isHttps
-      ? new HttpsAgent({
-          // we turn off normal CA cert validation when we are whitelisting a single cert thumbprint
-          rejectUnauthorized: options.acceptCertificate ? false : true,
-          // needed to allow whitelisting a cert thumbprint if a connection is reused
-          maxCachedSessions: options.acceptCertificate ? 0 : undefined,
-        })
-      : undefined,
-    maxRedirects: 0,
-  } as AxiosRequestConfig;
+    dispatcher: new ProxyAgent({
+      uri: options.proxy ? options.proxy : '',
+      maxRedirections: 0,
+      connect: {
+        rejectUnauthorized: options.acceptCertificate ? false : true,
+        maxCachedSessions: options.acceptCertificate ? 0 : undefined,
```

Comment:
We need to leave comments (that you removed) in our code, to have understanding of why it was configured